aloe/result.py

```diff
@@ -0,0 +1,175 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Cucumber-esque outputter for Nose.
+"""
+
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+from __future__ import absolute_import
+# pylint:disable=redefined-builtin, unused-wildcard-import, wildcard-import
+from builtins import *
+# pylint:enable=redefined-builtin, unused-wildcard-import, wildcard-import
+from future import standard_library
+standard_library.install_aliases()
+
+from contextlib import contextmanager
+
+from aloe.registry import (
+    CallbackDecorator,
+    CALLBACK_REGISTRY,
+    PriorityClass,
+)
+from aloe.strings import represent_table
+from blessings import Terminal
+from nose.result import TextTestResult
+
+# A decorator to add callbacks which wrap the steps looser than all the other
+# callbacks.
+# pylint:disable=invalid-name
+outer_around = CallbackDecorator(CALLBACK_REGISTRY, 'around',
+                                 priority_class=PriorityClass.DISPLAY)
+# pylint:enable=invalid-name
+#
+
+class StreamWrapper(object):
+    """Used to decorate file-like objects with a handy 'writeln' method"""
+    def __init__(self, stream):
+        self.stream = stream
+
+    def __getattr__(self, attr):
+        if attr in ('stream', '__getstate__'):
+            raise AttributeError(attr)
+        return getattr(self.stream,attr)
+
+    def write(self, arg=None, return_=False):
+        if not self.stream:
+            return
+
+        elif arg:
+            self.stream.write(arg)
+
+            if return_:
+                self.stream.write(term.move_up * arg.count('\n'))
+
+    def writeln(self, arg=None, return_=False):
+        if arg:
+            self.write(arg, return_=return_)
+
+        self.write('\n') # text-mode streams translate to \r\n if needed
+
+        if return_:
+            self.write(term.move_up)
+
+
+StreamWrapper = StreamWrapper(None)
+term = Terminal()
+
+
+@outer_around.each_feature
+@contextmanager
+def feature_wrapper(feature):
+
+    try:
+        if feature.tags:
+            StreamWrapper.writeln(term.cyan(feature.represent_tags()))
+
+        lines = feature.represented(annotate=False).splitlines()
+        StreamWrapper.writeln(term.bold_white(lines[0]))
+        StreamWrapper.writeln()
+        StreamWrapper.writeln(term.white('\n'.join(lines[1:])))
+        StreamWrapper.writeln()
+
+        yield
+    finally:
+        pass
```

Comment:
You would prefer I used before instead of around here? Seemed easier to only create the one priority decorator.